pkg/manager/config/config_test.go

```diff
@@ -153,8 +154,133 @@ func TestConfigRemove(t *testing.T) {
 	require.Eventually(t, func() bool { return cfgmgr.GetConfig().Proxy.Addr == "vv" }, time.Second, 100*time.Millisecond)
 }
 
+func TestFilePath(t *testing.T) {
+	var (
+		cfgmgr *ConfigManager
+		text   fmt.Stringer
+		count  int
+	)
+
+	tmpdir := t.TempDir()
+	checkLog := func(increased bool) {
+		// On linux, writing once will trigger 2 WRITE events. But on macOS, it only triggers once.
+		// So we always sleep 100ms to avoid missing any logs on the way.
+		time.Sleep(100 * time.Millisecond)
+		newCount := strings.Count(text.String(), "config file reloaded")
+		require.Equal(t, increased, newCount > count, fmt.Sprintf("now: %d, was: %d", newCount, count))
+		count = newCount
+	}
+
+	tests := []struct {
+		filename   string
+		createFile func()
+		cleanFile  func()
+		checker    func(filename string)
+	}{
+		{
+			// Test updating another file in the same directory won't make it reload.
+			filename: filepath.Join(tmpdir, "cfg"),
+			checker: func(filename string) {
+				tmplog := filepath.Join(tmpdir, "log")
+				f, err := os.Create(tmplog)
+				require.NoError(t, err)
+				require.NoError(t, f.Close())
+				require.NoError(t, os.WriteFile(tmplog, []byte("hello"), 0644))
+				newlog := filepath.Join(tmpdir, "log1")
+				require.NoError(t, os.Rename(tmplog, newlog))
+				require.NoError(t, os.Remove(newlog))
+				checkLog(false)
+			},
+		},
+		{
+			// Test case-insensitive.
+			filename: filepath.Join(tmpdir, "cfg"),
+			createFile: func() {
+				f, err := os.Create(filepath.Join(tmpdir, "CFG"))
+				require.NoError(t, err)
+				require.NoError(t, f.Close())
+				// Linux is case-sensitive but macOS is case-insensitive.
+				// For linux, it creates another file. For macOS, it doesn't touch the file.
+				f, err = os.Create(filepath.Join(tmpdir, "cfg"))
+				require.NoError(t, err)
+				require.NoError(t, f.Close())
+			},
+		},
+		{
+			// Test relative path.
+			filename: "cfg",
+		},
+		{
+			// Test relative path.
+			filename: "./cfg",
+		},
+		{
+			// Test uncleaned path.
+			filename: fmt.Sprintf("%s%c%ccfg", tmpdir, filepath.Separator, filepath.Separator),
+		},
+		{
+			// Test removing and creating the directory.
```

Comment:
There is a need to `removing and recreating the file`

Reply:
Addressed.